Readme.md

```diff
@@ -40,20 +40,20 @@ $sqsClient = new SqsClient([
 
 // Call a client's method with an array
 $result = $sqsClient->createQueue(['QueueName' => 'bar']);
-// Make sure the request is sent
-$result->resolve();
 
+// Request is automatically sent when reading the result
+echo $result->getMessageId();
 
 // You can also call a client's method with an input object
 $input = new SendMessageRequest();
 $input
     ->setQueueUrl('https://foo.com/bar')
     ->setMessageBody('foobar');
 
-$result = $sqsClient->sendMessage($input);
+// Since we dont use a $result to get the return value,
+// the HTTP request is sent automatically.
+$sqsClient->sendMessage($input);
```

Comment:
I want to show that we actually send the HTTP request on `__desctruct()`. 

A call to `resolve()` is rarely needed.

Reply:
it's not needed, but it helps troubleshouting exception. 
When `resolve` is called on destruct and trigger an exception, that sometime is hard to understand (from a user point of view)

Reply:
But in this case it is the exact opposite, right? 

Doing `$sqsClient->sendMessage($input);` will actually fire the request, If you get a 400 you will get an exception on this line. Ie, this is pretty much synchronous. 

The tricky part is when you do something like: 

```
function foo() {
   $result = $sqsClient->sendMessage($input);

  // Bunch of other code 

}
// the result will be destructed at the end of the block. 
``` 

But this is rare, either you need the result and use it, or you dont need it.

Reply:
My comment were just a reminder that `__destructor ` are hides sometime complexity in debugging.

Reply:
Yeah, true. We should document `resolve()` but not here in the readme.